modules/alloy/src/main/ts/ephox/alloy/positioning/view/Bounder.ts

```diff
@@ -11,146 +11,178 @@ import { SpotInfo } from './SpotInfo';
 export interface BounderAttemptAdt {
   fold: <T>(
     fit: (reposition: RepositionDecision) => T,
-    nofit: (reposition: RepositionDecision, deltaW: number, deltaH: number) => T
+    nofit: (reposition: RepositionDecision, visibleW: number, visibleH: number, isVisible: boolean) => T
   ) => T;
   match: <T>(branches: {
     fit: (reposition: RepositionDecision) => T;
-    nofit: (reposition: RepositionDecision, deltaW: number, deltaH: number) => T;
+    nofit: (reposition: RepositionDecision, visibleW: number, visibleH: number, isVisible: boolean) => T;
   }) => T;
   log: (label: string) => void;
 }
 
-interface RepositionResult {
+interface PositionResult {
   readonly originInBounds: boolean;
   readonly sizeInBounds: boolean;
-  readonly limitX: number;
-  readonly limitY: number;
-  readonly deltaW: number;
-  readonly deltaH: number;
+  readonly visibleW: number;
+  readonly visibleH: number;
+}
+
+interface MaxSizes {
+  readonly maxWidth: number;
+  readonly maxHeight: number;
 }
 
 const adt: {
   fit: (reposition: RepositionDecision) => BounderAttemptAdt;
-  nofit: (reposition: RepositionDecision, deltaW: number, deltaH: number) => BounderAttemptAdt;
+  nofit: (reposition: RepositionDecision, visibleW: number, visibleH: number, isVisible: boolean) => BounderAttemptAdt;
 } = Adt.generate([
   { fit: [ 'reposition' ] },
-  { nofit: [ 'reposition', 'deltaW', 'deltaH' ] }
+  { nofit: [ 'reposition', 'visibleW', 'visibleH', 'isVisible' ] }
 ]);
 
-const calcReposition = (newX: number, newY: number, width: number, height: number, bounds: Boxes.Bounds): RepositionResult => {
-  const boundsX = bounds.x;
-  const boundsY = bounds.y;
-  const boundsWidth = bounds.width;
-  const boundsHeight = bounds.height;
+/**
+ * This will attempt to determine if the box will fit within the specified bounds or if it needs to be repositioned.
+ * It will return the following details:
+ *  - if the original rect was in bounds (originInBounds & sizeInBounds). This is used to determine if we fitted
+ *    without having to make adjustments.
+ *  - the height and width deltas in relation to how much height/width would be visible in the original location.
```

Comment:
```suggestion
 *  - the width and height of the portion of the original rect that fits inside the bounds (may be negative if there is no overlap between the original rect and the bounds)
```

Reply:
That's not correct, it's literally the height/width delta not a portion of it.

Reply:
Isn't it just the size of the overlap?

Reply:
Yeah it is, but `portion` is the part I don't like/don't consider correct. The original wording is clearer for me at least.

Reply:
To me the original wording didn't really make sense, and I had to dig around to find out what these fields were. Most of that I think comes from calling them deltas, because it's not quite a difference between two things. Can we just describe it as:

```suggestion
 *  - the height and width that would be visible in the original location. (ie the overlap between the rect and the bounds)
```

Reply:
I really don't think describes the negative case well. It handles the positive case nicely though 🤷

Reply:
```suggestion
 *  - the height and width that would be visible in the original location. (ie the overlap between the rect and the bounds, may be negative if there is no overlap)
```

Reply:
That's still not ideal, but it's better than the original and conveys the negative value so lets go with that. Thanks!

Reply:
Done